sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -313,7 +379,7 @@ object BinaryArithmetic {
 case class Add(
     left: Expression,
     right: Expression,
-    failOnError: Boolean = SQLConf.get.ansiEnabled) extends BinaryArithmetic {
+    failOnError: Boolean = SQLConf.get.ansiEnabled) extends DecimalArithmetic {
```

Comment:
This looks a bit weird. I think supporting decimal is an additional ability, and we'd better use a mix-in trait, e.g. `extends BinaryArithmetic with DecimalArithmetic`

```
trait DecimalArithmetic { self: BinaryArithmetic 

}
```

Reply:
yeah, better to use a trait